code/modules/power/geothermal/_geothermal.dm

```diff
@@ -0,0 +1,103 @@
+var/global/const/GEOTHERMAL_EFFICIENCY_MOD =   0.2
+var/global/const/GEOTHERMAL_PRESSURE_LOSS =    0.3
+var/global/const/MAX_GEOTHERMAL_PRESSURE =    2000
+var/global/const/GEOTHERMAL_PRESSURE_TO_POWER = 100
+
+/obj/machinery/power/geothermal
+	icon = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state = "geothermal-base"
+	var/tmp/neighbors = 0
+	var/tmp/current_pressure = 0
+	var/tmp/processing = FALSE
+	var/efficiency = 0.5
+
+/obj/machinery/power/geothermal/RefreshParts()
+	..()
+	efficiency = Clamp(total_component_rating_of_type(/obj/item/stock_parts/capacitor) * GEOTHERMAL_EFFICIENCY_MOD, GEOTHERMAL_EFFICIENCY_MOD, 1)
+
+/obj/machinery/power/geothermal/proc/add_pressure(var/pressure)
+	current_pressure = Clamp(current_pressure + pressure, 0, MAX_GEOTHERMAL_PRESSURE)
+	if(!processing)
+		processing = TRUE
+		START_PROCESSING(SSmachines, src)
+
+/obj/machinery/power/geothermal/Process()
+	if(anchored && !(stat & BROKEN) && loc)
+		var/lost_pressure = current_pressure * 0.5
```

Comment:
this looks like a lot of calc but basic gist is that we 
1. Drop half of pressure no questions asked
2. Rest is distributed between lost/remaining based on efficiency

right?

Feels weird that we /always/ just toss out half of pressure, why not just have lower number in first place?
then could just have
```
lost_pressure = current_pressure * efficiency
remaining_pressure = current_pressure - lost_pressure 
```

Also this means that efficient turbines do not propogate anything down the chain, is that intended?

Reply:
No, we distribute half of our current pressure to our neighbors. Actual loss is calculated with GEOTHERMAL_PRESSURE_LOSS later. Process is:
- Take some portion of our total pressure.
- Turn some portion of that into power.
- Propagate whatever is left to our neighbors.

Totally efficient turbines eating the entire pressure chunk is fine imo.